bundler/lib/bundler/cli/gem.rb

```diff
@@ -180,7 +182,7 @@ def run
       if Bundler.git_present? && options[:git]
         Bundler.ui.info "Initializing git repo in #{target}"
         Dir.chdir(target) do
-          `git init`
+          `git init --initial-branch=#{git_default_branch}`
```

Comment:
My suggestion is that you don't pass `--initial-branch` at all. Instead, add `git rev-parse --abbrev-ref HEAD` after `git init` has run to check the name of the created branch, and fill in `git_default_branch` with that.

That we we don't have any backwards compatibility issues and we still respect user's choice.